senders/client_factory.go

```diff
@@ -44,6 +45,10 @@ type configuration struct {
 	// together with batch size controls the max theoretical throughput of the sender.
 	FlushIntervalSeconds int
 	SDKMetricsTags       map[string]string
+
+	Timeout int
```

Comment:
Is there any reason not to make this a `time.Duration`, matching the type that's used in `http.Client{}`? It certainly seems more expressive. Alternatively, name the field something like `TimeoutInSeconds`.

Reply:
Hi @oppegard, thanks for the feedback! Of the 2 options, I am leaning toward renaming the field to `TimeoutInSeconds`. 

Why: The caller of NewSender() can invoke a `TimeoutInSeconds()` setter with a `int` (rather than having to think about constructing `time.Duration`).  Then all they have to think about will be how long they want the timeout to be. 

However I do understand that this choice is inconsistent with the following theme: Reuse types used in `http.Client`. Especially because the TLSConfigOptions() exemplifies this theme. 

I'll check with `@LukeWinikates` before finalizing this change.

Reply:
@suprajanarasimhan I would say go with `time.Duration`. I agree it's slightly less convenient to create than an integer literal, but typically you would do something like `15 * time.Seconds` to create a `time.Duration`, so it ends up being very readable once you know about the pattern.

Reply:
Thank you!